Test Kitchen Integration

adamleff opened the pull request:
Supersedes #6.

Support for Test Kitchen integration/functional testing using new cisco-specific gems, and initial contribution of serverspec tests for current cookbook custom resources.

edolnx committed:
Adding first pass at Test Kitchen integration

edolnx committed:
Clarify documentation

edolnx committed:
re-enable package test, packages must be put in bootflash

Commit:
serverspec tests for the cisco cookbook

glennmatthews commented:
:+1: from me (the Rubocop failures look pretty trivial to fix post-merge) - Rob, if you're ok with it too, go ahead and merge.

robert-w-gries commented on test/README.md:
````diff
@@ -0,0 +1,31 @@
+## First run procedure
+
+ - Converge test cookbook to switch and verify using serverspec using ```chef exec bundle exec kitchen converge``` from the cookbook directory
+ - At this point, all tests should pass. You can cleanup by running ```chef exec bundle exec kitchen shutdown``` from the cookbook directory and removing /tmp/verifier directory on switch under test
````
This command didn't work for me.  Is it going to be implemented?

```
~/forks/chef-partners/cisco-network-chef-cookbook$ chef exec bundle exec kitchen shutdown
Could not find command "shutdown".
```

adamleff replied:
Actual command is `kitchen destroy` not "shutdown" - I will fix that.

robert-w-gries commented on test/README.md:
```diff
@@ -0,0 +1,31 @@
+ - ChefDK installed on local workstation
+ - Switch under test available via SSH
+ - Switch under test needs valid /etc/resolv.conf
+ - Switch under test needs Internet access (for rubygems) via some VRF
```
Can you add the following as a requirement?

`- Switch under test needs a user configured with bash login (Ex: username kitchen shelltype bash)`

adamleff replied:
You bet!

Commit:
README updates

Corrected a typo in the destroy vs. shutdown command, and added
an additional requirement per suggestion from the Cisco team.
